src/Domain/Contracts/Preset.php

```diff
@@ -6,25 +6,14 @@
 
 use NunoMaduro\PhpInsights\Application\Composer;
 
-/**
- * @internal
- */
 interface Preset
 {
-    /**
-     * Returns the preset name.
-     */
-    public static function getName(): string;
-
     /**
      * Returns the configuration preset.
      *
+     * @param \NunoMaduro\PhpInsights\Application\Composer $composer
```

Comment:
@olivernybroe We added this Composer param to check about php version constraint in DefaultPreset. 

As we now bumped to php7.4, do you think it's still relevant ? 

I'm afraid that this arguments "disturb" final users who want to create their own preset. WDYT ?

Reply:
Hmm, I think the scenario we needed for checking php7.4 will prob. also happen with PHP8.

Also even though we bumped to PHP7.4, I might use phpinsights on an older php version, if I for example used the docker image for running php insights.

However I get your point in it might be a little confusing for users why it's there. But I think it will be needed over time anyways.